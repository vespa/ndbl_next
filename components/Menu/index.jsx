import React from 'react';
import Link from 'next/link';

const Index = () => (
  <div>
    <Link href="/">
      <a href="/">
      Home
      </a>
    </Link>
    <Link href="/test">
      <a href="/test">About</a>
    </Link>
  </div>
);

export default Index;
