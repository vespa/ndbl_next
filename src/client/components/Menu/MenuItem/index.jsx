import React from 'react';
import Link from 'next/link';
import PropTypes from 'prop-types';
import cssMain from '../../../../scss/main.scss';

const MenuItem = ({ href, text }) => {
  return (
    <Link href={href}>
      <a href={href} className={`${cssMain.col_3}`}>
        {text}
      </a>
    </Link>
  );
};

MenuItem.propTypes = {
  href: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default MenuItem;
