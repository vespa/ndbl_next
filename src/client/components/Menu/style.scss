
@import '../../../scss/_vars.scss';

.main-menu {
  width: 100%;
  height: 3rem;
  background-color: $color_primary;
  text-transform: uppercase;
  font-weight: 700;
  position: fixed;
  top: 0;
  a {
    color: #fff;
    line-height: 3rem;
    text-align: center;
    transition: all 200ms;
    &:hover {
      background-color: #999;
      color: $color_primary
    }
    // display: block;
    // float: left;
  }
}