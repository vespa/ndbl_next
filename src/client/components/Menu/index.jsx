import React from 'react';
import cssMain from '../../../scss/main.scss';
import css from './style.scss';
import MenuItem from './MenuItem';
import SearchBox from '../SearchBox';
import MessageBox from '../MessageBox';


export const MenuOptions = () => (
  <React.Fragment>
    <MenuItem href="/" text="Home" />
    <MenuItem href="/about" text="About" />
  </React.Fragment>
);

export const Logo = () => (
  <div
    className={`
      ${css['main-menu__logo']}
    `}
  >
    <div
      className={`
        ${css['main-menu__logo__img']}
        ${cssMain.icon}
        ${cssMain.icon__logo}
      `}
    >
      Logo
    </div>
  </div>
);

const Menu = () => (
  <React.Fragment>
    <nav className={`${css['main-menu']}`}>
      <div className={`${cssMain.col_2}`}>
        <Logo />
      </div>
      <div className={cssMain.col_6}>
        <MenuOptions />
      </div>
      <div className={cssMain.col_3}>
        <SearchBox />
      </div>
      <div className={cssMain.col_1}>
        <MessageBox />
      </div>
    </nav>
    <nav className={`${css['main-menu-mobile']}`}>
      <div className={`${cssMain.m_col_4}`}>
        <button
          type="button"
          className={`
            ${css['main-menu-mobile__button']}
            ${cssMain.icon}
            ${cssMain.icon__menu}
          `}
        >
          Menu
        </button>
        <div className={`${css['main-menu-mobile__options']}`}>
          <MenuOptions />
        </div>
      </div>
      <div className={`${cssMain.m_col_4}`}>
        <Logo />
      </div>
    </nav>
  </React.Fragment>
);

export default Menu;
