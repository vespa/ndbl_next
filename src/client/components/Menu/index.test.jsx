import React from 'react';
import { shallow } from 'enzyme';
import Menu, { Logo, MenuOptions } from './index';

describe('[Component: Menu]', () => {
  let wrapper;

  it('should match snapshot', () => {
    wrapper = shallow(<Menu />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    wrapper = shallow(<Logo />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    wrapper = shallow(<MenuOptions />);
    expect(wrapper).toMatchSnapshot();
  });
});
