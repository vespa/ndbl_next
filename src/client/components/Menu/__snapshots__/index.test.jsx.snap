// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: Menu] should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <nav
          className="undefined"
        >
          <div
            className="undefined"
          >
            <Logo />
          </div>
          <div
            className={undefined}
          >
            <MenuOptions />
          </div>
          <div
            className={undefined}
          >
            <SearchBox />
          </div>
          <div
            className={undefined}
          >
            <MessageBox />
          </div>
        </nav>,
        <nav
          className="undefined"
        >
          <div
            className="undefined"
          >
            <button
              className="
            undefined
            undefined
            undefined
          "
              type="button"
            >
              Menu
            </button>
            <div
              className="undefined"
            >
              <MenuOptions />
            </div>
          </div>
          <div
            className="undefined"
          >
            <Logo />
          </div>
        </nav>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="undefined"
            >
              <Logo />
            </div>,
            <div
              className={undefined}
            >
              <MenuOptions />
            </div>,
            <div
              className={undefined}
            >
              <SearchBox />
            </div>,
            <div
              className={undefined}
            >
              <MessageBox />
            </div>,
          ],
          "className": "undefined",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Logo />,
              "className": "undefined",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MenuOptions />,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <SearchBox />,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MessageBox />,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="undefined"
            >
              <button
                className="
            undefined
            undefined
            undefined
          "
                type="button"
              >
                Menu
              </button>
              <div
                className="undefined"
              >
                <MenuOptions />
              </div>
            </div>,
            <div
              className="undefined"
            >
              <Logo />
            </div>,
          ],
          "className": "undefined",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="
            undefined
            undefined
            undefined
          "
                  type="button"
                >
                  Menu
                </button>,
                <div
                  className="undefined"
                >
                  <MenuOptions />
                </div>,
              ],
              "className": "undefined",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Menu",
                  "className": "
            undefined
            undefined
            undefined
          ",
                  "type": "button",
                },
                "ref": null,
                "rendered": "Menu",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <MenuOptions />,
                  "className": "undefined",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Logo />,
              "className": "undefined",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "nav",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <nav
            className="undefined"
          >
            <div
              className="undefined"
            >
              <Logo />
            </div>
            <div
              className={undefined}
            >
              <MenuOptions />
            </div>
            <div
              className={undefined}
            >
              <SearchBox />
            </div>
            <div
              className={undefined}
            >
              <MessageBox />
            </div>
          </nav>,
          <nav
            className="undefined"
          >
            <div
              className="undefined"
            >
              <button
                className="
            undefined
            undefined
            undefined
          "
                type="button"
              >
                Menu
              </button>
              <div
                className="undefined"
              >
                <MenuOptions />
              </div>
            </div>
            <div
              className="undefined"
            >
              <Logo />
            </div>
          </nav>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="undefined"
              >
                <Logo />
              </div>,
              <div
                className={undefined}
              >
                <MenuOptions />
              </div>,
              <div
                className={undefined}
              >
                <SearchBox />
              </div>,
              <div
                className={undefined}
              >
                <MessageBox />
              </div>,
            ],
            "className": "undefined",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Logo />,
                "className": "undefined",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MenuOptions />,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <SearchBox />,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MessageBox />,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="undefined"
              >
                <button
                  className="
            undefined
            undefined
            undefined
          "
                  type="button"
                >
                  Menu
                </button>
                <div
                  className="undefined"
                >
                  <MenuOptions />
                </div>
              </div>,
              <div
                className="undefined"
              >
                <Logo />
              </div>,
            ],
            "className": "undefined",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="
            undefined
            undefined
            undefined
          "
                    type="button"
                  >
                    Menu
                  </button>,
                  <div
                    className="undefined"
                  >
                    <MenuOptions />
                  </div>,
                ],
                "className": "undefined",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Menu",
                    "className": "
            undefined
            undefined
            undefined
          ",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Menu",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <MenuOptions />,
                    "className": "undefined",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Logo />,
                "className": "undefined",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "nav",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`[Component: Menu] should match snapshot 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Logo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="
        undefined
        undefined
        undefined
      "
      >
        Logo
      </div>,
      "className": "
      undefined
    ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Logo",
        "className": "
        undefined
        undefined
        undefined
      ",
      },
      "ref": null,
      "rendered": "Logo",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="
        undefined
        undefined
        undefined
      "
        >
          Logo
        </div>,
        "className": "
      undefined
    ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Logo",
          "className": "
        undefined
        undefined
        undefined
      ",
        },
        "ref": null,
        "rendered": "Logo",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`[Component: Menu] should match snapshot 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuOptions />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <MenuItem
          href="/"
          text="Home"
        />,
        <MenuItem
          href="/about"
          text="About"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "href": "/",
          "text": "Home",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "href": "/about",
          "text": "About",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MenuItem
            href="/"
            text="Home"
          />,
          <MenuItem
            href="/about"
            text="About"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "href": "/",
            "text": "Home",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "href": "/about",
            "text": "About",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
